File: symfony/notejam/src/AppBundle/Entity/Note.php
<?php

namespace AppBundle\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Note
 *
 * @ORM\Table()
 * @ORM\Entity(repositoryClass="AppBundle\Repository\NoteRepository")
 * @ORM\HasLifecycleCallbacks
 */
class Note
{
    /**
     * @var integer
     *
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    private $id;

    /**
     * @var string
     *
     * @ORM\Column(name="name", type="string", length=100)
     */
    private $name;

    /**
     * @var string
     *
     * @ORM\Column(name="text", type="text")
     */
    private $text;

    /**
     * @var datetime
     *
     * @ORM\Column(name="updated_at", type="datetime")
     */
    private $updated_at;

    /**
     * @var User
     *
     * @ORM\ManyToOne(targetEntity="\AppBundle\Entity\User", inversedBy="notes")
     * @ORM\JoinColumn(name="user_id", referencedColumnName="id")
     */
    private $user;

    /**
     * @var User
     *
     * @ORM\ManyToOne(targetEntity="\AppBundle\Entity\Pad", inversedBy="notes")
     * @ORM\JoinColumn(name="pad_id", referencedColumnName="id")
     */
    private $pad;


    /**
     * Get id
     *
     * @return integer
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * Set name
     *
     * @param string $name
     * @return Note
     */
    public function setName($name)
    {
        $this->name = $name;

        return $this;
    }

    /**
     * Get name
     *
     * @return string
     */
    public function getName()
    {
        return $this->name;
    }

    /**
     * Set text
     *
     * @param string $text
     * @return Note
     */
    public function setText($text)
    {
        $this->text = $text;

        return $this;
    }

    /**
     * Get text
     *
     * @return string
     */
    public function getText()
    {
        return $this->text;
    }

    /**
     * Get user
     *
     * @return User
     */
    public function getUser()
    {
        return $this->user;
    }


    /**
     * Set user
     *
     * @param User user
     * @return Note
     */
    public function setUser($user)
    {
        $this->user = $user;
        return $this;
    }

    /**
     * Get pad
     *
     * @return Pad
     */
    public function getPad()
    {
        return $this->pad;
    }


    /**
     * Set pad
     *
     * @param Pad pad
     * @return Note
     */
    public function setPad($pad)
    {
        $this->pad = $pad;
        return $this;
    }

    /**
     * Update timestamp
     *
     * @ORM\PrePersist
     * @ORM\PreUpdate
     */
    public function setUpdatedAt()
    {
        $this->updated_at = new \DateTime('now');
    }

    /**
     * Get last update time
     *
     * @return string formatted date
     */
    public function getUpdatedAt()
    {
        return $this->updated_at->format('d M. Y');
    }

    /**
     * Get name
     *
     * @return string
     */
    public function __toString() {
        return $this->getName();
    }
}

File: notejam/spring/src/main/java/net/notejam/spring/note/Note.java
package net.notejam.spring.note;

import java.time.Instant;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Index;
import javax.persistence.Lob;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

import org.hibernate.validator.constraints.NotEmpty;
import org.springframework.data.jpa.domain.AbstractPersistable;

import net.notejam.spring.pad.Pad;
import net.notejam.spring.security.owner.Owned;
import net.notejam.spring.user.User;

/**
 * The note.
 *
 * @author markus@malkusch.de
 * @see <a href="bitcoin:1335STSwu9hST4vcMRppEPgENMHD2r1REK">Donations</a>
 */
@Entity
@Table(indexes = { @Index(columnList = "updated"), @Index(columnList = "name") })
public class Note extends AbstractPersistable<Integer>implements Owned {

    private static final long serialVersionUID = -1445367127777923455L;

    /**
     * The last update time.
     */
    @NotNull
    private Instant updated;

    /**
     * The owner.
     */
    @ManyToOne
    @NotNull
    private User user;

    /**
     * The pad.
     */
    @ManyToOne
    private Pad pad;

    /**
     * The name.
     */
    @NotEmpty
    @Size(max = 100)
    private String name;

    /**
     * The text.
     */
    @NotEmpty
    @Lob
    @Column(length = 10000)
    private String text;

    /**
     * Returns the name.
     *
     * @return The name.
     */
    public String getName() {
        return name;
    }

    /**
     * Sets the name.
     *
     * @param name The name.
     */
    public void setName(final String name) {
        this.name = name;
    }

    /**
     * Returns the text.
     *
     * @return The text
     */
    public String getText() {
        return text;
    }

    /**
     * Sets the text.
     *
     * @param text The text.
     */
    public void setText(final String text) {
        this.text = text;
    }

    /**
     * Returns the pad.
     *
     * @return The pad or null.
     */
    public Pad getPad() {
        return pad;
    }

    /**
     * Sets the pad.
     *
     * @param pad The pad or null.
     */
    public void setPad(final Pad pad) {
        this.pad = pad;
    }

    /**
     * Sets the owner.
     *
     * @param user The owner.
     */
    public void setUser(final User user) {
        this.user = user;
    }

    @Override
    public User getUser() {
        return user;
    }

    /**
     * Returns the last update time.
     *
     * @return The time.
     */
    public Instant getUpdated() {
        return updated;
    }

    /**
     * Sets the last update time.
     *
     * @param updated The time.
     */
    public void setUpdated(final Instant updated) {
        this.updated = updated;
    }

}
